<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:arg name="link1_x" default="0.076"/>
    <xacro:arg name="link1_y" default="0"/>
    <xacro:arg name="link1_z" default="0.076"/>
    <xacro:arg name="link1_joint_x" default="0.04"/>
    <xacro:arg name="link1_joint_z" default="0.008"/>

    <link name="gripper_link_base">
        <inertial>
            <mass value="0.05"/>
            <origin rpy="1.571 0 0" xyz="0 0 0" />
            <inertia ixx="0.02216" ixy="5.8e-06" ixz="6.2e-06" iyy="0.02"
                iyz="-5.3e-05"
                izz="0.02"/>
        </inertial>
        <collision>
            <origin rpy="1.571 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="file:///$(find arm_03)/description/meshes/gripA/gripA_0.stl" 
                    scale="0.004 0.004 0.004" />                
            </geometry>
        </collision>
        <visual>
            <origin rpy="1.571 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="file:///$(find arm_03)/description/meshes/gripA/gripA_0.stl" 
                    scale="0.004 0.004 0.004" />
            </geometry>
            <material name="yellow">
                <color rgba="1 1 0 1"/>
            </material>                  
        </visual>
    </link>

    <link name="gripper_link_left_1">
        <contact>
            <plugin name="gripper_link_left_1_contact_plugin" filename="libgazebo_ros_contact.so"/>
        </contact>    
        <inertial>
            <mass value="0.05"/>
            <origin xyz="4e-06   -0.001     0.082"/>
            <inertia ixx="3.9" ixy="-0.11" ixz="-8e-9" iyy="0.6" iyz="7e-9" izz="3.7"/>
        </inertial>
        <collision>
            <origin rpy="1.571 0 0" xyz="-$(arg link1_x) $(arg link1_y) $(arg link1_z)" />
            <geometry>
                <mesh filename="file:///$(find arm_03)/description/meshes/gripA/gripA_L_1.stl" 
                    scale="0.004 0.004 0.004" />
                <!-- cylinder radius="0.08" length="0.2"/ -->
            </geometry>
        </collision>            
        <visual>
            <origin rpy="1.571 0 0" xyz="-$(arg link1_x) $(arg link1_y) $(arg link1_z)" />
            <!-- origin rpy="1.571 0 0" xyz="-0.036 0 0.038" /-->
            <geometry>
                <mesh filename="file:///$(find arm_03)/description/meshes/gripA/gripA_L_1.stl" 
                    scale="0.004 0.004 0.004" />
            </geometry>
            <material name="red">
                <color rgba="1 0 0 1"/>
            </material>                  
        </visual>

        <surface>
            <friction>
                <ode>
                    <mu>1.0</mu>
                    <mu2>1.0</mu2>
                </ode>
            </friction>
            <contact>
                <ode>
                    <kp>100</kp>
                    <kd>100</kd>
                    <min_depth>0.005</min_depth>
                    <max_vel>0.0</max_vel>
                </ode>
            </contact>
        </surface>        

    </link>        

    <joint name="gripper_joint_left_1" type="prismatic">
        <parent link="gripper_link_base" />
        <child link="gripper_link_left_1" />
        <origin rpy="0 0 0" xyz="-$(arg link1_joint_x) $(arg link1_y) $(arg link1_joint_z)" />
        <!-- origin rpy="0 0 0" xyz="-0.036 0 -0.04" / -->
        <axis xyz="1 0 0" />
        <limit effort="100" lower="-0.04" upper="0.1" velocity="0.05"/>
        <dynamics damping="60.0" friction="4.0" />
    </joint>     

    <link name="gripper_link_right_1">
        <contact>
            <plugin name="gripper_link_right_1_contact_plugin" filename="libgazebo_ros_contact.so"/>
        </contact>    
        <inertial>
            <mass value="0.05"/>
            <origin xyz="4e-06   -0.001     0.082"/>
            <inertia ixx="3.9" ixy="0.1" ixz="8e-9" iyy="0.6" iyz="7e-9" izz="3.7"/>
        </inertial>
        <collision>
            <origin rpy="1.571 0 0" xyz="$(arg link1_x) $(arg link1_y) $(arg link1_z)" />
            <geometry>
                <!-- cylinder radius="0.08" length="0.2"/ -->
                <mesh filename="file:///$(find arm_03)/description/meshes/gripA/gripA_R_1.stl" 
                    scale="0.004 0.004 0.004" />
            </geometry>
        </collision>            
        <visual>
            <origin rpy="1.571 0 0" xyz="$(arg link1_x) $(arg link1_y) $(arg link1_z)" />
            <!-- origin rpy="1.571 0 0" xyz="-0.036 0 0.038" /-->
            <geometry>
                <mesh filename="file:///$(find arm_03)/description/meshes/gripA/gripA_R_1.stl" 
                    scale="0.004 0.004 0.004" />
            </geometry>
            <material name="red">
                <color rgba="1 0 0 1"/>
            </material>                  
        </visual>

        <surface>
            <friction>
                <ode>
                    <mu>1.0</mu>
                    <mu2>1.0</mu2>
                </ode>
            </friction>
            <contact>
                <ode>
                    <kp>100</kp>
                    <kd>100</kd>
                    <min_depth>0.005</min_depth>
                    <max_vel>0.0</max_vel>
                </ode>
            </contact>
        </surface>        

    </link>        

    <joint name="gripper_joint_right_1" type="prismatic">
        <parent link="gripper_link_base" />
        <child link="gripper_link_right_1" />
        <origin rpy="0 0 0" xyz="$(arg link1_joint_x) $(arg link1_y) $(arg link1_joint_z)" />
        <!-- origin rpy="0 0 0" xyz="-0.036 0 -0.04" / -->
        <axis xyz="1 0 0" />
        <limit effort="100" lower="-0.1" upper="0.04" velocity="0.05"/>
        <dynamics damping="60.0" friction="4.0" />

        <!-- Mimic the movement of gripper_joint_left_1 -->
        <!-- mimic joint="gripper_joint_left_1" multiplier="-1" offset="0" / -->

    </joint>

    <!-- Gazebo colors -->

    <gazebo reference="gripper_link_base">
        <material>Gazebo/Yellow</material>
    </gazebo>
    <gazebo reference="gripper_link_left_1">
        <material>Gazebo/Red</material>
    </gazebo>     
    <gazebo reference="gripper_link_right_1">
        <material>Gazebo/Blue</material>
    </gazebo>           
    

</robot>
